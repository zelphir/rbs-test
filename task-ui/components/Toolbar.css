.toolbar {
  width: 100%;
  height: 60px;
  background: #000;
  position: fixed;
  top: 0;
  display: flex;
}

.toolbar input {
  flex: 1;
  margin: 10px;
  border: 0;
  outline: 0;
  font-size: 18px;
  padding: 5px;
  font-weight: 100;
  background: rgba(255, 255, 255, 0.7);
}

.toolbar .sort {
  color: rgba(255, 255, 255, 0.7);
  flex: 0 0 25%;
  text-align: right;
  line-height: 40px;
  padding: 10px;
}
