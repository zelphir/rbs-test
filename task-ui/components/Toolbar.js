import React from 'react'
import PropTypes from 'prop-types'

import './Toolbar.css'

const Toolbar = ({ searchBy, sortBy }) => {
  return (
    <div className="toolbar">
      <input placeholder="Search by name..." onChange={searchBy} />
      <div className="sort">
        Sort by Amount:{' '}
        <button onClick={sortBy} data-order="ASC">
          ASC
        </button>{' '}
        |{' '}
        <button onClick={sortBy} data-order="DESC">
          DESC
        </button>
      </div>
    </div>
  )
}

Toolbar.propTypes = {
  searchBy: PropTypes.func.isRequired,
  sortBy: PropTypes.func.isRequired
}

export default Toolbar
