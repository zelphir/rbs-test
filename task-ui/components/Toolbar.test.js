import React from 'react'
import { shallow } from 'enzyme'
import Toolbar from './Toolbar'

const searchBy = jest.fn()
const sortBy = jest.fn()

describe('<Toolbar />', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Toolbar searchBy={searchBy} sortBy={sortBy} />)
    expect(wrapper).toMatchSnapshot()
  })
})
