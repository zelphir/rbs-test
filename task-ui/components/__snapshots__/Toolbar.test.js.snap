// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> should match the snapshot 1`] = `
<div
  className="toolbar"
>
  <input
    onChange={[MockFunction]}
    placeholder="Search by name..."
  />
  <div
    className="sort"
  >
    Sort by Amount:
     
    <button
      data-order="ASC"
      onClick={[MockFunction]}
    >
      ASC
    </button>
     
    |
     
    <button
      data-order="DESC"
      onClick={[MockFunction]}
    >
      DESC
    </button>
  </div>
</div>
`;
